config/jobs/kubernetes-csi/gen-jobs.sh

```diff
@@ -131,22 +131,55 @@ resources_for_kubernetes () {
     case $kubernetes in master|$experimental_k8s_version|release-*) cat <<EOF
 resources:
   requests:
-    # these are both a bit below peak usage during build
-    # this is mostly for building kubernetes
-    memory: "9000Mi"
-    # during the tests more like 3-20m is used
-    cpu: 2000m
+    memory: "12Gi"
+    cpu: 6
+  limits:
+    memory: "12Gi"
+    cpu: 6
 EOF
                             ;;
                             *) cat <<EOF
 resources:
   requests:
-    cpu: 2000m
+    memory: "9Gi"
+    cpu: 4
+  limits:
+    memory: "9Gi"
+    cpu: 4
 EOF
                             ;;
     esac | sed -e "s/^/${indentation}/"
 }
 
+job_cluster() {
+  local repo=$1
+
+  # Declare an associative array
+  declare -A cluster_mapping
+
+  cluster_mapping["eks-prow-build-cluster"]="external-snapshotter external-resizer"
```

Comment:
This seems complicated. Wouldn't it be simpler to do a switch?
```
case "$repo" in
   external-snapshotter|external-resizer) echo "eks-prow-build-cluster";;
   *) echo "default"
```

Reply:
Sure, I can update it to a case. Personally, I think the array structure will be more clean when dealing with all of the repos under csi.

Reply:
The variable may be simpler, but the shell code to work with it isn't.

Reply:
Fair enough, I'll update it and add in the generated files in the next few hours.